src/transformation/context/context.ts

```diff
@@ -39,7 +39,8 @@ export class TransformationContext {
     public readonly luaTarget = this.options.luaTarget ?? LuaTarget.Universal;
     public readonly isModule = ts.isExternalModule(this.sourceFile);
     public readonly isStrict =
-        (this.options.alwaysStrict ?? this.options.strict) ||
+        this.options.alwaysStrict ??
+        this.options.strict ??
         (this.isModule && this.options.target !== undefined && this.options.target >= ts.ScriptTarget.ES2015);
```

Comment:
This looks like it has a different meaning, before if alwaysStrict and strict were both false, but the latter clause is true, the result would be true, now if alwaysStrict is not undefined but false, it will not even look at the final clause

Reply:
I'm not really sure, this is what the ESLint rule auto-fixed to, so I assume that it is safe. If you are worried about it, I can revert the change and then ignore the ESLint error, just let me know.

Reply:
It's not the same, this should be reverted. Maybe if you change `(this.options.alwaysStrict ?? this.options.strict)` to `(this.options.alwaysStrict ?? this.options.strict ?? false)` it will swallow it without needing an eslint ignore, or else maybe pulling that out to a separate variable might do the trick

Reply:
ok its reverted now